R-package/R/lightgbm.R

```diff
@@ -173,6 +182,17 @@ lightgbm <- function(data,
     , alternative_kwarg_value = verbose
   )
 
+  # Process factors as labels and auto-determine objective
+  if (!lgb.is.Dataset(data)) {
+    data_processor <- DataProcessor$new()
+    temp <- data_processor$process_y(label, objective, params)
+    label <- temp$y
+    objective <- temp$objective
+    params <- temp$params
```

Comment:
```suggestion
    params <- temp$params
    rm(temp)
```

Once the code is done extract what it needs from this object, please remove it from memory. Since it contains a copy of the label, it could occupy a meaningful amount of memory.